persist/fs/msgpack/encoder.go

```diff
@@ -140,13 +157,37 @@ func (enc *Encoder) EncodeLogMetadata(entry schema.LogMetadata) error {
 }
 
 func (enc *Encoder) encodeIndexInfo(info schema.IndexInfo) {
+	enc.encodeRootObject(indexInfoVersion, indexInfoType)
+	if enc.encodeLegacyV1IndexInfo {
+		enc.encodeIndexInfoV1(info)
+	} else {
+		enc.encodeIndexInfoV2(info)
+	}
+}
+
+// We only keep this method around for the sake of testing
+// backwards-compatbility
+func (enc *Encoder) encodeIndexInfoV1(info schema.IndexInfo) {
```

Comment:
could you check if these are being inlined?

Reply:
do you mean inlined by the compiler for performance reasons? This method is so far outside the hot-path I don't think there is any value in that

Reply:
Agreed, for info files this should be fine, it's called once per minute at most =]